client/engine/chainservice/eth_chainservice.go

```diff
@@ -116,84 +119,105 @@ func (ecs *EthChainService) SendTransaction(tx protocols.ChainTransaction) error
 		return fmt.Errorf("unexpected transaction type %T", tx)
 	}
 }
-func (ecs *EthChainService) subcribeToEvents() error {
 
-	go ecs.listenForLogEvents()
-	return nil
+// dispatchChainEvents takes in a collection of event logs from the chain
+// and dispatches events to the out channel
+func (ecs *EthChainService) dispatchChainEvents(logs []ethTypes.Log) {
+	for _, l := range logs {
+		switch l.Topics[0] {
+		case depositedTopic:
+			nad, err := ecs.na.ParseDeposited(l)
+			if err != nil {
+				ecs.logger.Printf("error in ParseDeposited: %v", err)
+			}
+
+			event := NewDepositedEvent(nad.Destination, l.BlockNumber, nad.Asset, nad.AmountDeposited, nad.DestinationHoldings)
+			ecs.out <- event
+		case allocationUpdatedTopic:
+			au, err := ecs.na.ParseAllocationUpdated(l)
+			if err != nil {
+				ecs.logger.Printf("error in ParseAllocationUpdated: %v", err)
+			}
+
+			tx, pending, err := ecs.chain.TransactionByHash(context.Background(), l.TxHash)
```

Comment:
This fails (pretty silently for FEVM) causing the problems with decoding the transaction. The problem seems to be the transaction we get back from wallaby doesn't have a signature:
```
 {"jsonrpc":"2.0","id":108,"result":{"chainId":"0x7ab7","nonce":"0x0","hash":"0x8bd654ecaabd9b43540c13ad2e640907ff4a55d2ded44053e6fd59e8bd568c2f","blockHash":"0x79c4060be1b7763c504a6b765cdbc19214aebece4ae58c30ce84d72346ca5a53","blockNumber":"0x7f43","transacionIndex":"0x0","from":"0xd41298190d223f90c9475aa98852d764ed53550d","to":"0x9f6e424d79cea7e32b852f724e130d2f1c7c99c6","value":"0x0","type":"0x2","input":"0x18025<SNIPPED>","gas":"0x3a164da","maxFeePerGas":"0x494a8","maxPriorityFeePerGas":"0x493e0","v":"0x","r":"0x","s":"0x"}}
```

This results in the `go-ethereum` transaction serialisation failing with this error:
```
error in TransactoinByHash: json: cannot unmarshal hex string "0x" into Go struct field txJSON.v of type *hexutil.Big
```

Reply:
I've created an [issue](https://github.com/filecoin-project/ref-fvm/issues/1158) in the fevm repo.